web/src/settings_org.ts

```diff
@@ -754,24 +756,212 @@ export function deactivate_organization(e: JQuery.Event): void {
+
+    function delete_data_in_text(): string {
+        const $delete_in = $<HTMLSelectOneElement>("select:not([multiple])#delete-realm-data-in");
+        const delete_data_value = $delete_in.val()!;
+
+        if (delete_data_value === "null") {
+            return $t({defaultMessage: "Data will not be automatically deleted"});
+        }
+
+        let time_in_minutes: number;
+        if (delete_data_value === "custom") {
+            if (!util.validate_custom_time_input(custom_deletion_time_input)) {
+                return $t({defaultMessage: "Invalid custom time"});
+            }
+            time_in_minutes = util.get_custom_time_in_minutes(
+                custom_deletion_time_unit,
+                custom_deletion_time_input,
+            );
+            if (!is_valid_time_period(time_in_minutes)) {
+                return $t({defaultMessage: "Invalid custom time"});
+            }
+        } else {
+            // These options were already filtered for is_valid_time_period.
+            time_in_minutes = Number.parseFloat(delete_data_value);
+        }
+
+        if (time_in_minutes === 0) {
+            return $t({defaultMessage: "Data will be deleted immediately"});
+        }
+
+        // The below is a duplicate of timerender.get_full_datetime, with a different base string.
+        const valid_to = add(new Date(), {minutes: time_in_minutes});
+        const date = timerender.get_localized_date_or_time_for_format(valid_to, "dayofyear_year");
+        return $t({defaultMessage: "Data will be deleted after {date}"}, {date});
+    }
+
+    const minimum_allowed_days = realm.server_min_deactivated_realm_deletion_days ?? 0;
+    const maximum_allowed_days = realm.server_max_deactivated_realm_deletion_days;
+
+    function is_valid_time_period(time_period: string | number): boolean {
+        if (time_period === "custom") {
+            return true;
+        }
+        if (time_period === "null") {
+            if (maximum_allowed_days === null) {
+                return true;
+            }
+            return false;
+        }
+        if (typeof time_period === "number") {
+            if (maximum_allowed_days === null) {
+                if (time_period >= minimum_allowed_days * 24 * 60) {
+                    return true;
+                }
+            } else {
+                if (
+                    time_period >= minimum_allowed_days * 24 * 60 &&
+                    time_period <= maximum_allowed_days * 24 * 60
+                ) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+    function get_custom_deletion_input_text(): string {
+        if (maximum_allowed_days === null) {
+            return $t({defaultMessage: `Custom time ({min}+ days)`}, {min: minimum_allowed_days});
+        }
+        return $t(
+            {defaultMessage: `Custom time ({min}-{max} days)`},
+            {min: minimum_allowed_days, max: maximum_allowed_days},
+        );
+    }
```

Comment:
This seems to not handle the cases that the minimum is null?

Reply:
Tested it displays 0+, which seems OK.

Reply:
Added this while I was here:

``` diff
diff --git a/web/src/settings_org.ts b/web/src/settings_org.ts
index 4f544d8d65..865d891c92 100644
--- a/web/src/settings_org.ts
+++ b/web/src/settings_org.ts
@@ -857,6 +857,12 @@ export function deactivate_organization(e: JQuery.Event): void {
 
     function get_custom_deletion_input_text(): string {
         if (maximum_allowed_days === null) {
+            if (minimum_allowed_days == 0) {
+                // If there's no limit at all, avoid showing 0+. It's
+                // not a marginal string for translators, since we use
+                // that string elsewhere.
+                return $t({defaultMessage: "Custom time"});
+            }
             return $t({defaultMessage: `Custom time ({min}+ days)`}, {min: minimum_allowed_days});
         }
         return $t(
```